ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -66,18 +66,24 @@
   private final QueryId queryId;
   private final QueryErrorClassifier errorClassifier;
   private final Queue<QueryError> queryErrors;
+  private final RetryEvent retryEvent;
 
   private Optional<QueryStateListener> queryStateListener = Optional.empty();
   private boolean everStarted = false;
   protected boolean closed = false;
-  private UncaughtExceptionHandler uncaughtExceptionHandler = this::uncaughtHandler;
+  private StreamsUncaughtExceptionHandler uncaughtExceptionHandler = this::uncaughtHandler;
   private KafkaStreams kafkaStreams;
   private Consumer<Boolean> onStop = (ignored) -> { };
   private boolean initialized = false;
 
+  private static final Ticker CURRENT_TIME_MILLIS_TICKER = new Ticker() {
+    @Override
+    public long read() {
+      return System.currentTimeMillis();
+    }
+  };
+
   // CHECKSTYLE_RULES.OFF: ParameterNumberCheck
-  @VisibleForTesting
```

Comment:
Is not just used in testing. Maybe vestigial or I am misunderstanding the tag?

Reply:
I think the annotation generally means that something is public/package-private instead of private so we can use it outside the class. So if it's used in testing at all, this still applies

Reply:
alright